tests/integration/mirrorpeersecret_controller_test.go

```diff
@@ -0,0 +1,615 @@
+					var inNamespace client.InNamespace = client.InNamespace(eachPeer.ClusterName)
+					err := k8sClient.List(ctx, &sourceList, hasSourceLabel, inNamespace)
+					Expect(err).To(BeNil())
+					Expect(sourceList.Items).NotTo(BeEmpty())
+					Expect(len(sourceList.Items) == 1).To(BeTrue())
+					sourceSecret := sourceList.Items[0]
+					connectedPeers := controllers.PeersConnectedToPeerRef(eachPeer, mpList)
+					for _, eachConnectedPeer := range connectedPeers {
+						destNN := types.NamespacedName{Namespace: eachConnectedPeer.ClusterName, Name: sourceSecret.Name}
+						var destSecret corev1.Secret
+						err := k8sClient.Get(ctx, destNN, &destSecret)
+						Expect(err).To(BeNil())
+						Expect(destSecret.Data[common.SecretDataKey]).To(BeEquivalentTo(sourceSecret.Data[common.SecretDataKey]))
+						Expect(destSecret.Labels).To(HaveKey(common.SecretLabelTypeKey))
+						Expect(destSecret.Labels[common.SecretLabelTypeKey]).To(BeEquivalentTo(common.DestinationLabel))
+					}
+				}
+			})
+
+			By("should remove any orphan destination secrets", func() {
+				time.Sleep(500 * time.Millisecond)
+				var mpList multiclusterv1alpha1.MirrorPeerList
+				err := k8sClient.List(ctx, &mpList)
+				Expect(err).To(BeNil())
+
+				var destList corev1.SecretList
+				var hasDestinationLabel client.MatchingLabels = make(client.MatchingLabels)
+				hasDestinationLabel[common.SecretLabelTypeKey] = string(common.DestinationLabel)
+				err = k8sClient.List(ctx, &destList, hasDestinationLabel)
+				Expect(err).To(BeNil())
+				for _, eachDestSecret := range destList.Items {
+					peerRef, err := common.CreatePeerRefFromSecret(&eachDestSecret)
+					Expect(err).To(BeNil())
+					connectedPeers := controllers.PeersConnectedToPeerRef(peerRef, mpList.Items)
+					Expect(connectedPeers).NotTo(BeEmpty())
+					var sourceListOptions []client.ListOption
+					for _, eachConnectedPeer := range connectedPeers {
+						var inNamespace client.InNamespace = client.InNamespace(eachConnectedPeer.ClusterName)
+						sourceListOptions = append(sourceListOptions, inNamespace)
+					}
+					var hasSourceLabel client.MatchingLabels = make(client.MatchingLabels)
+					Expect(err).To(BeNil())
+					hasSourceLabel[common.SecretLabelTypeKey] = string(common.SourceLabel)
+
+					sourceListOptions = append(sourceListOptions, hasSourceLabel)
+					var allConnectedSources corev1.SecretList
+					err = k8sClient.List(ctx, &allConnectedSources, sourceListOptions...)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(allConnectedSources.Items).NotTo(BeEmpty())
+					var oneOfTheNameShouldMatch bool
+					for _, eachConnectedSource := range allConnectedSources.Items {
+						if eachConnectedSource.Name == eachDestSecret.Name {
+							oneOfTheNameShouldMatch = true
+							break
+						}
+					}
+					Expect(oneOfTheNameShouldMatch).To(BeTrue())
+				}
+			})
+		})
+	})
+
+	When("test ends", func() {
+		It("should remove all the resources", func() {
```

Comment:
Should be handled in `After*` block.

Reply:
Same as the above comment... Finally instead of `Before*` and `After*` I went in a straight direction (from start to end).